frontend/src/pages/EditUserPage.tsx

```diff
@@ -0,0 +1,273 @@
+
+  // Arrays below are used to add the role/agency from the dropdown later in code
+  let agenciesToUpdate: any[] = user.agencies;
+  let rolesToUpdate: any[] = user.roles;
+
+  const initialValues = {
+    username: user.username,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    email: user.email,
+    displayName: user.displayName,
+    isDisabled: false,
+    rowVersion: user.rowVersion,
+    emailVerified: false,
+    agencies: user.agencies,
+    roles: user.roles,
+    note: user.note,
+    agency: '',
+    role: '',
+  };
+
+  return (
+    <div>
+      <Navbar className="navBar" expand="sm" variant="light" bg="light">
+        <Navbar.Brand href="#">User Information</Navbar.Brand>
+      </Navbar>
+      <Container fluid={true}>
+        <Row className="justify-content-md-center">
+          <Formik
+            enableReinitialize
+            initialValues={initialValues}
+            validationSchema={UserUpdateSchema}
+            onSubmit={(values, { setSubmitting }) => {
+              if (values.agency !== '') {
+                agenciesToUpdate.push({ id: Number(values.agency) });
+              }
+
+              if (values.role !== '') {
+                rolesToUpdate.push({ id: values.role });
+              }
+
+              dispatch(
+                getUpdateUserAction(
+                  { id: '41a46ea3-e815-4272-9a17-693ba2409841' },
+                  {
+                    id: user.id,
+                    username: user.username,
+                    displayName: values.displayName,
+                    firstName: values.firstName,
+                    lastName: values.lastName,
+                    email: values.email,
+                    isDisabled: values.isDisabled,
+                    rowVersion: values.rowVersion,
+                    emailVerified: values.emailVerified,
+                    agencies: agenciesToUpdate,
+                    roles: rolesToUpdate,
+                  },
+                ),
+              );
+              setSubmitting(false);
+              alert('Refresh to see changes take effect on UI');
+            }}
+          >
+            {props => (
+              <Form className="userInfo">
+                <FormGroup controlId="displayName">
+                  <FormLabel>IDIR/BCeID</FormLabel>
+                  <FormControl
+                    readOnly={true}
+                    type="text"
+                    className="usernameField"
+                    value={props.values.username}
+                  />
+                </FormGroup>
+
+                <FormGroup controlId="firstName">
+                  <Row>
+                    <Col>
+                      <FormLabel>First Name</FormLabel>
+                      <FormControl
+                        type="text"
+                        className="firstName"
+                        placeholder={props.values.firstName}
+                        onChange={(e: any) => (props.values.firstName = e.target.value)}
```

Comment:
This is a great start for this form! 

Just a couple observations to help improve it:

- You are almost there by using "controlled form components" (definition [here](https://reactjs.org/docs/forms.html#controlled-components)). However, you shouldn't assign directly a new value to the props like above.
- It should be more like `<FormControl ... value={props.values.firstName} onChange={props.handleChange}`
- Note that I typed `onChange={props.handleChange}` <-- the handleChange method is provided by  **Formik** to keep the form state updated.

For my map filter PR, I took some time to create Formik-connected base components; i.e. `<Form>`, `<Input>`, `<Select>`, `<Button>` to help with future form development. 

Please take a look under `src/components/common/form/` - granted there will be missing pieces there (like `<Textarea>` as I didn't need it on my form - but hopefully the rest you can re-use. 

I'll try to cover Form Development on our next dev session to cover these ideas/patterns with the team.

Reply:
Sweet, thank you for the tips! I'll start adjusting this form accordingly. And sounds good re form development, that would be helpful!